app/javascript/components/bookmarks/BookmarkManager.jsx

```diff
@@ -86,11 +89,64 @@ export default function BookmarkManager({bookmarks, studyAccession, clearExplore
     setCurrentBookmark(findExistingBookmark())
   }
 
-  /** whenever the user changes a cluster/annotation, reset the bookmark form with the current URL params */
+  /** close open form if changing tabs */
+  function closeBookmarkForm() {
+    if (isUserLoggedIn() && formRef.current?.state?.show) {
+      formRef.current.handleHide()
+    }
+  }
+
+  /** wrapper to close both bookmark form & modal */
+  function closeAllComponents() {
+    closeBookmarkForm()
```

Comment:
I wonder if it'd be possible to pass in a triggering event for logging, e.g. "escape", "outside-keydown".  This is merely food for thought for later, and not blocking.